### PR TITLE
feat: refactor, patch to v1 account info

### PR DESCRIPTION
**Pull request by evilpeach**
## Describe your changes
- Implement v1 account info that consists of ICNS, account public info, and public project info.
- Apply Zod to query params.

## Demo Link
[Demo](https://celatone-frontend-staging-git-feat-cfe-64-acco-7bdcf4-alleslabs.vercel.app/osmosis-1/accounts/osmo18kmnpjw6mj7juw6wmnzdyaa8e2tg9h3mqry0ym)

### CODE REVIEW

File: src/lib/types/projects.ts [hunk 1]
```diff
@@ -100,3 +104,32 @@ export interface PublicInfo {
...
+  socials: z.array(
+    z.object({
+      name: z.string(),
+      url: z.string(),
+    })
+  ),
+  website: z.string(),
+});
+
+export type ProjectInfoV1 = z.infer<typeof zProjectInfo>;
```
Comment (songwongtp) [inline src/lib/types/projects.ts:125]: remove V1

File: src/lib/types/projects.ts [hunk 2]
```diff
@@ -100,3 +104,32 @@ export interface PublicInfo {
...
+
+export const zPublicAccount = z.object({
+  address: zAddr,
+  description: z.string(),
+  name: z.string(),
+  slug: z.string(),
+  type: z.string(),
+});
+
+export type PublicAccountV1 = z.infer<typeof zPublicAccount>;
```
Comment (songwongtp) [inline src/lib/types/projects.ts:135]: ditto

File: src/lib/types/projects.ts [hunk 3]
```diff
@@ -100,3 +104,32 @@ export interface PublicInfo {
   description: string;
   github: string;
 }
+
+// ------------------------------------------//
+// --------------------V1--------------------//
+// ------------------------------------------//
+export const zProjectInfo = z.object({
```
Comment (songwongtp) [inline src/lib/types/projects.ts:111]: remove export

File: src/lib/types/projects.ts [hunk 4]
```diff
@@ -100,3 +104,32 @@ export interface PublicInfo {
...
+      name: z.string(),
+      url: z.string(),
+    })
+  ),
+  website: z.string(),
+});
+
+export type ProjectInfoV1 = z.infer<typeof zProjectInfo>;
+
+export const zPublicAccount = z.object({
```
Comment (songwongtp) [inline src/lib/types/projects.ts:127]: also remove export

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/types/projects.ts (hunk 3 above)
Comment (evilpeach) [reply src/lib/types/projects.ts:111]: I used it in `services/account.ts`.

File: src/lib/types/projects.ts (hunk 4 above)
Comment (evilpeach) [reply src/lib/types/projects.ts:127]: This also used in `services/account.ts`.

File: src/lib/pages/account-details/components/AccountHeader.tsx [hunk 5]
```diff
@@ -164,7 +161,9 @@ export const AccountHeader = observer(
                     }}
                     gap={1}
                   >
-                    {name === icnsName.primary_name && <PrimaryNameMark />}
+                    {name === accountInfo?.icns?.primary_name && (
```
Comment (songwongtp) [inline src/lib/pages/account-details/components/AccountHeader.tsx:164]: can we remove `?` ?

File: src/lib/pages/account-details/index.tsx [hunk 6]
```diff
@@ -447,24 +438,26 @@ const AccountDetails = () => {
...
     <PageContainer>
-      {!isSomeValidAddress(accountAddressParam) ? (
+      {!isSomeValidAddress(accountAddress) ? (
         <InvalidAccount />
       ) : (
         <AccountDetailsBody
-          accountAddressParam={accountAddressParam}
-          tab={tab}
+          accountAddressParam={accountAddress}
+          tabParam={tab}
```
Comment (songwongtp) [inline src/lib/pages/account-details/index.tsx:460]: change props to `accountAddress` and `tab` as they are not from query params any more, then we can just `...validated.data` here

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Review by songwongtp — APPROVED** [review]

**Pull request merged**